helix-lsp/src/client.rs

```diff
@@ -25,6 +30,17 @@ use tokio::{
     },
 };
 
+fn workspace_for_uri(uri: lsp::Url) -> WorkspaceFolder {
+    lsp::WorkspaceFolder {
+        name: uri
+            .path_segments()
+            .and_then(|segments| segments.last())
+            .map(|basename| basename.to_string())
```

Comment:
What do we do if there are conflicting workspace names?

Reply:
Workspace names are unimportant. RA for example.ignores them completely (most LS do). The language standard recommends using the base name (what we do here) as a default. LS only use these names in some messages: Like `indexing foo`